docs/blueprints/linux-setup.md

```diff
@@ -0,0 +1,187 @@
+# Storage node installation and update on Linux
+
+## Abstract
+
+The design document outlines the steps needed to ease installation and provide auto-update of storage nodes on Linux without the use of Docker.
+
+
+## Background
+
+
+The idea is to have two main components, like for Windows: the storage node binary and a storage node updater that will take care of updating the storage node binary according to the rollout version mechanism.
+
+The parts we need:
+- storagenode as a service
+- a system for updating the storagenode binary aka. the updater (with rollout versioning support)
+- a system for updating the updater 
+- a way to collect the configuration data from the user during the installation
+- packaging to ship the above
+
+Linux systems are not unified as windows is. Hence, we have to make choices that will allow us to cover most linux users, with special regards to the raspbian system.
+
+Most commonly used linux distributions seem to be (there are no reliable stats):
+- debian-based (Debian, Ubuntu, Mint, Kali, Raspbian)
+- Red Hat-based (Fedora)
+- Arch Linux (Manjaro is on the rise according to [distrowatch](https://distrowatch.com/dwres.php?resource=popularity)).
+
+All these distributions are shipped with systemd as a service manager.
+
+
+
+
+## Design
+The installer will be a debian package. We choose to auto-update the binary, even if this will make the package not following debian guidelines
+- When installing the package, it should prompt for user input for:
+  - Wallet address
+  - Email
+  - External address/port
+  - Advertised storage
+  - Identity directory 
+  - Storage directory
+- Generate `config.yaml` file with the user configuration.
+
+The default value for these directories can be defined using the [XDG Base Directory](https://wiki.archlinux.org/index.php/XDG_Base_Directory).
+
+We choose to reuse the storagenode-updater and the recovery mechanism used in windows. They will be daemonized using systemd. The storagenode updater will auto-update. A recovery will be triggered if the updated updater service fails to restart.
+We will use debconf to retrieve user data. 
+
+The debian package will NOT contain the storagenode and storagenode-updater binaries. They will be downloaded as part of the post-installation script. A separate git repository will be created for holding the debian package.
+
+Once we get a fully working debian package, we can convert it to the RPM format using the fpm tool. There are no debconf-like for RPMs, we will need to implement a post-install script to gather user inputs.
+
+The debian package will be available by direct download and on a APT repository that users can add to their package manager source list. The repository will be managed using reprepro. Each time the repository is modified, it commits the static content to a dedicated git repository. 
```

Comment:
It might be worth defining in this blueprint where the repository signing key will be stored, how access to it is managed, and how they are to use it when updating the repo. I'm not sure what we do for Windows, but the situation will probably be similar?